### DISCUSSION TITLE
Access: allow AccessSQL only

### DISCUSSION THREAD

**Discussion by 2019-05-10** [dbeaver/dbeaver#14329] [Q&A] [answered]
Currently I am trying to develop queries for an Access 2013 DB.
While those queries work well in DBeaver, they often do not when pasted in Acess.

After some research I learned that Access on top of all its other shortcomings does not support all SQL features DBeaver (or rather UCanaccess) allows.
Additionally, it requires an inordinate and rarely intuitive amount of braces to be put elsewhere.

So, if possible, it would be of great help to let DBeaver's SQL editor accept only that subset of SQL Access natively accepts (and maybe even propose the braces - there seem in fact to be rules for those, though I fail to grasp them).
That way I could extensively check queries and simply copy them over to Access once finished.

**Comment by uslss** [comment]
thanks for suggestion

**Comment by serge-rider** [comment, marked as answer]
Unfortunately we can't control this as Access SQL dialect is provided by UCanAccess driver. It is based on hsqldb engine as far as I know.

Alternatively you could use Access ODBC driver.

Also there are several other commercial Access drivers which use Access engine directly so there shouldn't be any problems with SQL dialect compatibility.
